Spigot-API-Patches/0057-Add-UnknownCommandEvent.patch

```diff
@@ -0,0 +1,85 @@
+From 3395e2bb0176e4be7bbe798ee9eebf67204f9b24 Mon Sep 17 00:00:00 2001
+From: Sweepyoface <github@sweepy.pw>
+Date: Wed, 14 Jun 2017 11:12:01 -0400
+Subject: [PATCH] Add UnknownCommandEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/command/UnknownCommandEvent.java b/src/main/java/org/bukkit/event/command/UnknownCommandEvent.java
+new file mode 100644
+index 00000000..cdd349c3
+--- /dev/null
++++ b/src/main/java/org/bukkit/event/command/UnknownCommandEvent.java
+@@ -0,0 +1,70 @@
++package org.bukkit.event.command;
++
++import org.bukkit.command.CommandSender;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.Event;
++
++/**
++ * Thrown when a player executes a command that is not defined
++ */
++public class UnknownCommandEvent extends Event {
```

Comment:
I'd just suggest using a null to allow nothing to be sent. As I believe some other events do that when they can't be cancelled. Whether you'd want to notion of this event being cancellable is another question. You can't logically cancel an the command being unknown >_>